prov/gni/src/gnix_ep.c

```diff
@@ -360,15 +359,12 @@ static void __gnix_vc_destroy_ht_entry(void *val)
  * EP vc initialization helper
  ******************************************************************************/
 
-static inline int __gnix_ep_init_vc(struct gnix_fid_ep *ep_priv)
+int _gnix_ep_init_vc(struct gnix_fid_ep *ep_priv, enum fi_av_type type)
```

Comment:
If every instance uses ep->av->type, we can retrieve that value once in the function rather than passing it as a parameter. Putting it on the stack here would have merit if we expected the av type to change during execution AND we wanted to preserve the value through the execution of the function. Neither of those cases seem to be true here.